browser/src/react-redux/App/LoginPage.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import logInUser from './logInUser.action'
+import axios from 'axios'
+
+const initialState = {
+  email: '',
+  password: '',
+  generalError: ''
+}
+
+class LoginPage extends React.Component {
+  state = initialState
+
+  handleChange = event => {
+    this.setState({
+      [event.target.name]: event.target.value
+    })
+  }
+
+  onSubmit = e => {
+    this.setState(initialState)
+
+    e.preventDefault()
```

Comment:
Let's toss this at the top. If we're ever dealing with a bug with `onSubmit()` that is a result of an error happening within the function, if it's before the moment we try to prevent the default behavior, the uncaught error can prevent `preventDefault()` from actually being executed. This will result in a bug that isn't easily detectable because we're rerouted instantly.

Just an example of how this can happen: We convert this function, for whatever reason, to a normal function, and then accidentally lose appropriate context to this. `this.setState()` would actually now give us the error that we can no longer call setState on undefined (or something similar).